scipy/stats/_qmc.py

```diff
@@ -1319,47 +1323,56 @@ class Sobol(QMCEngine):
     """
 
     MAXDIM: ClassVar[int] = _MAXDIM
-    MAXBIT: ClassVar[int] = _MAXBIT
 
     def __init__(
-            self, d: IntNumber, *, scramble: bool = True,
-            seed: SeedType = None
+        self, d: IntNumber, *, scramble: bool = True,
+        bits: IntNumber = 30, seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         if d > self.MAXDIM:
             raise ValueError(
-                "Maximum supported dimensionality is {}.".format(self.MAXDIM)
+                f"Maximum supported dimensionality is {self.MAXDIM}."
             )
 
+        self.MAXBIT = bits
+        if self.MAXBIT > 64:
+            raise ValueError("Maximum supported 'bits' is 64")
+        self.MAXN = 2**self.MAXBIT
+
         # initialize direction numbers
-        initialize_direction_numbers()
+        _initialize_direction_numbers()
```

Comment:
I understand that "The underscore is for some reason making the Cython code better supported by PyCharm." This is fine (but again, if this is orthogonal to adding `bits`, I'd suggest it be in a separate PR).